daffodil-codegen-c/src/main/scala/org/apache/daffodil/codegen/c/DaffodilCCodeGenerator.scala

```diff
@@ -259,6 +260,7 @@ object DaffodilCCodeGenerator
   def generateCode(gram: Gram, cgState: CodeGeneratorState): Unit = {
     gram match {
       case g: AlignmentFill => alignmentFillGenerateCode(g, cgState)
+      case g: AssertBooleanPrim => noop(g)
```

Comment:
We have a schema with assertions which we want daffodil to evaluate for the additional validation they perform.  They aren't meant to cause any backtracking, only print diagnostics (their failureType is "recoverableError").  At the same time, we want daffodilC to ignore these assertions silently since we would need to support more of the DFDL expression language than the generator currently supports.

Yes, it could be a good idea to add code to print a warning if the assertion does not have failureType="recoverableError", which would mean the schema might be using the assertion for backtracking.  That will have to wait for another time since it's non-trivial to find the right symbols to inspect.

Reply:
That's fine, not critical for this PR.